### PR TITLE
Remove hardcoded values for subsections

### PR DESCRIPTION
**Pull request by andresmgot**
Ref: #1182

Follow up of #1223 

This PR allow developers to use specific keys to render subsections in a generic way. If a parameter enables or disables a different subsection, that parameter is moved as a children of the subsection. Examples:

 - `mariadb.enabled` disables `externalDatabase`:

```json
    "mariadb": {
      "type": "object",
      "properties": {
        "enabled": {
          "type": "boolean",
          "form": "useSelfHostedDatabase",
          "disable": "externalDatabase"
        }
      }
    },
    "externalDatabase": {
      "type": "object",
      "title": "External Database Details",
      "form": "externalDatabase",
      "properties": {
        "host": {
          ...
```

 - `ingress.enabled` enables `ingress`:

```json
    "ingress": {
      "type": "object",
      "form": "ingress",
      "properties": {
        "enabled": {
          "type": "boolean",
          "form": "enableIngress",
          "enable": "ingress",
        },
        "hostname": {
          "type": "string",
          "form": "hostname",
          "title": "Hostname"
        }
      }
    },
```

### CODE REVIEW

File: dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/BasicDeploymentForm.tsx [hunk 1]
```diff
@@ -100,6 +53,19 @@ class BasicDeploymentForm extends React.Component<IBasicDeploymentFormProps> {
...
+                label={param.title || name}
+                handleValuesChange={this.props.handleValuesChange}
+                appValues={this.props.appValues}
+                renderParam={this.renderParam}
+                key={id}
+                name={name}
+                param={param}
+              />
+            );
+          }
```
Comment (absoludity) [inline dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/BasicDeploymentForm.tsx]: Nice - getting simpler each time (DRYing up) :)

File: dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/Subsection.test.tsx [hunk 2]
```diff
@@ -28,6 +28,7 @@ const defaultProps = {
         title: "Enable External Database",
         type: "boolean",
         value: true,
+        disable: "externalDatabase",
```
Comment (absoludity) [inline dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/Subsection.test.tsx]: It'd be a tiny bit clearer if this was `disables: "externalDatabase"` - as in, this param disables the `externalDatabase` param (`disable` without the `s` could mean disable this current param, or disable the following param).

File: dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/Subsection.tsx [hunk 3]
```diff
@@ -23,9 +21,34 @@ export interface ISubsectionProps {
...
+  enablerCondition: boolean;
+}
+
+function findEnabler(name: string, param: IBasicFormParam) {
+  let result = { enablerChildrenParam: "", enablerCondition: false };
+  const children = param.children;
+  if (children) {
+    Object.keys(children).forEach(p => {
+      if (children[p].type === "boolean") {
+        if (children[p].enable === name) {
```
Comment (absoludity) [inline dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/Subsection.tsx]: Same here - `enables` would be a tad clearer in intent.

File: dashboard/src/shared/schema.test.ts [hunk 4]
```diff
@@ -188,6 +188,82 @@ externalDatabase:
...
+            foo: { path: "externalDatabase.host", type: "string" },
+            enableDB: {
+              path: "external.database",
+              type: "boolean",
+              enable: "externalDatabase",
+            },
+          },
+        } as IBasicFormParam,
+      },
+    },
```
Comment (absoludity) [inline dashboard/src/shared/schema.test.ts]: Excellent clear tests!

File: dashboard/src/shared/schema.ts [hunk 5]
```diff
@@ -11,13 +11,6 @@ import { IBasicFormParam } from "./types";
 const { nullOptions } = require("yaml/types");
 nullOptions.nullStr = "";
 
-// Form keys that require pre-definition. This list should be kept as small as possible
-export const EXTERNAL_DB = "externalDatabase";
-export const USE_SELF_HOSTED_DB = "useSelfHostedDatabase";
-export const RESOURCES = "resources";
-export const INGRESS = "ingress";
-export const ENABLE_INGRESS = "enableIngress";
```
Comment (absoludity) [inline dashboard/src/shared/schema.ts]: Woohoo - all gone :)

File: dashboard/src/shared/schema.ts [hunk 6]
```diff
@@ -67,15 +60,24 @@ export function retrieveBasicFormParams(
 }
 
 // orderParams conveniently structure the parameters to satisfy a parent-children relationship even if
-// those parameters doesn't have that relation in the source
+// those parameters doesn't have that relation in the source. This is only used when a parameter
```
Comment (absoludity) [inline dashboard/src/shared/schema.ts]: s/doesn't/don't/

File: dashboard/src/shared/schema.ts [hunk 7]
```diff
@@ -67,15 +60,24 @@ export function retrieveBasicFormParams(
...
+// enables/disables another.
+// CAVEAT: It only works with one level of depth
```
Comment (absoludity) [inline dashboard/src/shared/schema.ts]: Looks fixable in the future if/when we need to (depending on what we find in charts). Thanks for the note.

File: dashboard/src/shared/types.ts [hunk 8]
```diff
@@ -387,6 +387,8 @@ export interface IBasicFormParam {
   sliderMin?: number;
   sliderMax?: number;
   sliderUnit?: string;
+  disable?: string;
+  enable?: string;
```
Comment (absoludity) [inline dashboard/src/shared/types.ts]: Similar to my comment in the last PR, not sure if `IBasicFormEnableDisableParam` composing `IBasicFormParam` would mean we don't have to mix in slider-specific and enable/disable specific fields here (and in the future).

Comment (andresmgot) [reply dashboard/src/shared/types.ts]: as I said, I am not sure that would really improve the code but let me send a follow up PR and we can discuss that there.

**Comment by andresmgot** [comment]
Thanks for the review!

**Pull request merged**